python/guide/packages/gluon/data.md

```diff
@@ -1,325 +1,120 @@
 
-```{.python .input}
-dogcat.synsets
-```
+#### Why would you want to do this?
 
-Next let's print a particular sample with its label:
+Most of the time, samples returned by the dataset aren't in quite the right format to be passed into a neural network. One common example is to have samples with a data type of `int8` when your network expects `float32`. Another common example with images is when you have an image with a dimension order of (height, width, channel) when you need (channel, height, width) for the network.
 
-```{.python .input}
-sample = dogcat[1]
-plt.imshow(sample[0].asnumpy())
-plt.show()
-'label:', dogcat.synsets[sample[1]]
-```
+Augmenting samples is also common practice: that is, randomly permuting existing samples to make new samples to reduce issues of network overfitting. Using image samples as an example, you could crop to random regions, flip from left to right or even jitter the brightness of the image.
 
-## Transform Data Examples
+#### And how do you do it?
 
-The raw data examples often need to be transformed
-before feeding into a neural network. Class `Dataset` provides two methods
-`transform` and `transform_first` to allow users to specify the transformation
-methods.
+We first define our transform and then apply it to our dataset.
 
-In the following example, we define a function to resize an image into
-200px height and 300px width. And then we pass it into the dataset through the
-`transform_first` method, which returns a new dataset with the transformations
-recorded.
+MXNet Gluon has a number of in-build transforms available at `mxnet.gluon.data.vision.transforms`, so let's use [`RandomResizedCrop`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.vision.transforms.RandomResizedCrop.html) as an example. We're specifying that we want a random crop that contains 60% to 80% of the original image, and then we want to scale this cropped region to 20px by 20px. You should notice we're instantiating a class that can be called like a function.
 
-```{.python .input}
-def resize(x):
-    y = image.imresize(x, 300, 200)
-    print('resize', x.shape, 'into', y.shape)
-    return y
-dogcat_resized = dogcat.transform_first(resize)
 ```
+from mxnet.gluon.data.vision.transforms import RandomResizedCrop
 
-As can be seen, transformations are applied when accessing examples, which is
-necessary when these transformations contain randomness. But we can also apply
-all transformations during creating the dataset.
-
-```{.python .input}
-dogcat_cached = dogcat.transform_first(resize, lazy=False)
+transform_fn = RandomResizedCrop(size=(20, 20), scale=(0.6, 0.8))
 ```
 
-So no transform is needed during accessing examples.
+After defining `transform_fn`, we now need to apply it to the dataset. We call the `transform_first` method of our dataset to apply `transform_fn` to the first element of all samples. We had 2 elements per sample in our example, so we only apply `transform_fn` to the image and not the label.
```

Comment:
I would add a short sentence on the fact, that it is a lazy transformation.

Reply:
Done, a bit further down.